src/PolygonDraw/usePolygonEditor.ts

```diff
@@ -0,0 +1,100 @@
+import { useReducer, useEffect, useMemo, useCallback, useRef } from 'react';
+import { Coordinate } from 'types';
+import isEqual from 'lodash.isequal';
+
+import { actions } from './actions';
+import { ensurePolygonList, isPolygonClosed } from '../helpers';
+import { polygonEditReducer, initialState } from './reducer';
+import { isValidPolygon } from './validators';
+
+function usePrevious(value: Coordinate[][]) {
+    const ref = useRef<Coordinate[][]>();
+    useEffect(() => {
+        ref.current = value;
+    }, [value]);
+    return ref.current;
+}
+
+export const usePolygonEditor = (
+    onChange: (polygon: Coordinate[], isValid: boolean) => void = () => {},
+    polygons: Coordinate[] | Coordinate[][],
+    activeIndex: number
+) => {
+    const polygonList = ensurePolygonList(polygons);
+
+    const [state, dispatch] = useReducer(polygonEditReducer, initialState, init => {
+        return {
+            ...init,
+            activeIndex: activeIndex,
+            polygons: polygonList
+        };
+    });
```

Comment:
Using the reducer state results in redundant renders since the changes are propagated to the container and back to the component once again.
Since the PolygonDraw is a controlled component by design, I think we should use the previous approach. Later we can think about either transforming it to be uncontrolled or create a different one.

Reply:
Done 👍